contracts/ethregistrar/ETHRegistrarController.sol

```diff
@@ -1,178 +1,299 @@
 
-        if(msg.value > cost) {
-            payable(msg.sender).transfer(msg.value - cost);
+        if (msg.value > basePrice) {
+            payable(msg.sender).transfer(msg.value - basePrice);
         }
 
-        emit NameRenewed(name, label, cost, expires);
+        emit NameRenewed(name, label, basePrice, expires);
     }
 
     function setPriceOracle(PriceOracle _prices) public onlyOwner {
         prices = _prices;
         emit NewPriceOracle(address(prices));
     }
 
-    function setCommitmentAges(uint _minCommitmentAge, uint _maxCommitmentAge) public onlyOwner {
-        minCommitmentAge = _minCommitmentAge;
-        maxCommitmentAge = _maxCommitmentAge;
+    // function setCommitmentAges(
+    //     uint256 _minCommitmentAge,
+    //     uint256 _maxCommitmentAge
+    // ) public onlyOwner {
+    //     minCommitmentAge = _minCommitmentAge;
+    //     maxCommitmentAge = _maxCommitmentAge;
+    // }
+
+    function withdraw() public {
+        payable(owner()).transfer(address(this).balance);
     }
 
-    function withdraw() public onlyOwner {
-        payable(msg.sender).transfer(address(this).balance);        
+    function supportsInterface(bytes4 interfaceID)
+        external
+        pure
+        returns (bool)
+    {
+        return
+            interfaceID == INTERFACE_META_ID ||
+            interfaceID == COMMITMENT_CONTROLLER_ID;
     }
 
-    function supportsInterface(bytes4 interfaceID) external pure returns (bool) {
-        return interfaceID == INTERFACE_META_ID ||
-               interfaceID == COMMITMENT_CONTROLLER_ID ||
-               interfaceID == COMMITMENT_WITH_CONFIG_CONTROLLER_ID;
+    function onERC1155Received(
+        address,
+        address,
+        uint256,
+        uint256,
+        bytes calldata
+    ) external pure returns (bytes4) {
+        //TODO: guard against any other contract sending erc1155 to it apart from NameWrapper
+        return
+            bytes4(
+                keccak256(
+                    "onERC1155Received(address,address,uint256,uint256,bytes)"
+                )
+            );
     }
 
-    function _consumeCommitment(string memory name, uint duration, bytes32 commitment) internal returns (uint256) {
+    /* Internal functions */
+
+    function _consumeCommitment(
+        string memory name,
+        uint256 duration,
+        bytes32 commitment
+    ) internal {
         // Require a valid commitment
         require(commitments[commitment] + minCommitmentAge <= block.timestamp);
 
         // If the commitment is too old, or the name is registered, stop
         require(commitments[commitment] + maxCommitmentAge > block.timestamp);
         require(available(name));
 
-        delete(commitments[commitment]);
+        delete (commitments[commitment]);
 
-        uint cost = rentPrice(name, duration);
         require(duration >= MIN_REGISTRATION_DURATION);
-        require(msg.value >= cost);
```

Comment:
why do we get rid of this check?

Reply:
We get rid of this check to change the logic of the contract to just register how much they sent in ETH. You can see we check a new function from the oracle called `duration` which allows us to figure out how long they want to register for